packages/components/src/components/dialog/dialog.jsx

```diff
@@ -6,126 +6,122 @@ import { CSSTransition } from 'react-transition-group';
-                    appear: 'dc-dialog__wrapper--enter',
-                    enter: 'dc-dialog__wrapper--enter',
-                    enterDone: 'dc-dialog__wrapper--enter-done',
-                    exit: 'dc-dialog__wrapper--exit',
-                }}
-                unmountOnExit
+    const dialog = (
+        <CSSTransition
+            appear
+            in={is_visible && !is_loading}
+            timeout={50}
+            classNames={{
+                appear: 'dc-dialog__wrapper--enter',
+                enter: 'dc-dialog__wrapper--enter',
+                enterDone: 'dc-dialog__wrapper--enter-done',
+                exit: 'dc-dialog__wrapper--exit',
+            }}
+            unmountOnExit
+        >
+            <div
+                className={classNames('dc-dialog__wrapper', className, {
+                    'dc-dialog__wrapper--has-portal': !!portal_element_id,
+                })}
             >
                 <div
-                    className={classNames('dc-dialog__wrapper', className, {
-                        'dc-dialog__wrapper--has-portal': !!portal_element_id,
+                    className={classNames('dc-dialog__dialog', {
+                        'dc-dialog__dialog--has-margin': !is_mobile_full_width,
                     })}
                 >
-                    <div
-                        className={classNames('dc-dialog__dialog', {
-                            'dc-dialog__dialog--has-margin': !is_mobile_full_width,
-                        })}
-                    >
-                        <div className='dc-dialog__header-wrapper'>
-                            {!!title && <h1 className='dc-dialog__header'>{title}</h1>}
-                            {has_close_icon && (
-                                <div
-                                    onClick={onCancel ? this.handleCancel : this.handleConfirm}
-                                    className='dc-dialog__header--close'
-                                >
-                                    <Icon icon='IcCross' />
-                                </div>
-                            )}
-                        </div>
-                        {typeof children === 'string' ? (
-                            <p className={content_classes}>{children}</p>
-                        ) : (
-                            <div className={content_classes}>{children}</div>
+                    <div className='dc-dialog__header-wrapper'>
+                        {!!title && <h1 className='dc-dialog__header'>{title}</h1>}
+                        {has_close_icon && (
+                            <div onClick={onCancel ? handleCancel : handleConfirm} className='dc-dialog__header--close'>
+                                <Icon icon='IcCross' />
+                            </div>
+                        )}
+                    </div>
+                    {typeof children === 'string' ? (
+                        <p className={content_classes}>{children}</p>
```

Comment:
just a question.
Can we use Text component here?

Reply:
Yes, let's do it in a separate PR though together with other `p` to `Text` changes.